src/plugins/discover/public/plugin.tsx

```diff
@@ -375,7 +375,11 @@ export class DiscoverPlugin
         const services = await getDiscoverServicesInternal();
         const initialState = await deserializeState({
           serializedState: {
-            rawState: { savedObjectId: savedObject.id },
+            rawState: savedObject.managed
+              ? {
+                  attributes: { ...savedObject.attributes, references: savedObject.references },
+                }
+              : { savedObjectId: savedObject.id },
```

Comment:
I got an error when testing. Can it be improved?

<img width="770" alt="Screenshot 2024-08-13 at 11 06 18" src="https://github.com/user-attachments/assets/ed4f1c75-c639-4369-8bc7-7b22d21c9732">

Reply:
Can you give some repro instructions?

Reply:
Ah sorry. I imported this file
[dashboard.ndjson.zip](https://github.com/user-attachments/files/16603556/dashboard.ndjson.zip) and it looked like some references were not resolved to actually render the dashboard. Then I duplicated the dashboard. And the search panel showed this message.
I still have the instance running https://jughosta-pr-190202-embeddablerebuild-fix-managed-2024-08-08.kbndev.co/app/dashboards#/view/a1e918f4-4ca9-4ee9-8e6f-fc5971eb70d9?_g=(filters:!(),refreshInterval:(pause:!t,value:60000),time:(from:now-15m,to:now))

Reply:
We are exploring a potential fix here: https://github.com/elastic/kibana/pull/190555